src/pages/Login/Login.page.jsx

```diff
@@ -1,38 +1,154 @@
+  flex-direction: column;
+  align-items: center;
+  background-color: ${(props) => props.theme.highBackground};
+  vertical-align: middle;
+  box-shadow: 0 0 15px 3px rgba(0, 0, 0, 0.26);
+  border-radius: 18px;
+  padding: 30px;
+  width: 100%;
+  max-width: 400px;
+  min-height: 400px;
+`;
+
+const ErrorLabel = styled.h4`
+  color: red;
+`;
+
+const LoginPageContainer = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  width: 100%;
+  min-height: 91vh;
+  margin-top: 5px;
+  padding-left: 3px;
+  text-align: center;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+`;
 
 function LoginPage() {
   const { login } = useAuth();
   const history = useHistory();
+  const inputUserName = useRef(null);
+  const inputPassword = useRef(null);
+  const [userName, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [errorLogin, setErrrorLogin] = useState(false);
+
+  function onChangeUserName(value) {
+    setUsername(() => value);
+  }
+
+  function onChangePassword(value) {
+    setPassword(() => value);
+  }
 
-  function authenticate(event) {
+  async function authenticate(event) {
     event.preventDefault();
-    login();
-    history.push('/secret');
+    try {
+      await login(userName, password);
+      history.push('/favorites');
+    } catch (error) {
+      setErrrorLogin(true);
+    }
   }
 
   return (
-    <section className="login">
-      <h1>Welcome back!</h1>
-      <form onSubmit={authenticate} className="login-form">
-        <div className="form-group">
-          <label htmlFor="username">
-            <strong>username </strong>
-            <input required type="text" id="username" />
-          </label>
-        </div>
-        <div className="form-group">
-          <label htmlFor="password">
-            <strong>password </strong>
-            <input required type="password" id="password" />
-          </label>
-        </div>
-        <button type="submit">login</button>
-      </form>
-    </section>
+    <LoginPageContainer>
+      <Login>
+        <H1>Welcome back!</H1>
+        <LoginForm onSubmit={authenticate}>
+          <Div>
+            <FieldLabel htmlFor="username">username </FieldLabel>
+            <Input
+              ref={inputUserName}
+              required
+              type="text"
+              id="username"
+              onChange={(event) => onChangeUserName(event.target.value)}
+            />
+          </Div>
+          <Div>
+            <FieldLabel htmlFor="password">password </FieldLabel>
+            <Input
+              ref={inputPassword}
+              required
+              type="password"
+              id="password"
+              onChange={(event) => onChangePassword(event.target.value)}
+            />
+          </Div>
+          {errorLogin ? (
+            <>
+              <ErrorLabel>Incorrect user or password try again</ErrorLabel>
+            </>
+          ) : (
+            <> </>
+          )}
```

Comment:
You could return null instead of an empty fragment

Reply:
done thanks